### DISCUSSION TITLE
wiki/G__GraphicalEditor

### DISCUSSION THREAD

**Discussion by giscus** [gama-platform/gama-platform.github.io#199] [General]
# wiki/G__GraphicalEditor

The graphical editor allows defining a GAMA model through a graphical interface (gadl files). It is based on the Graphiti Eclipse plugin. It allows as well to produce a graphical model (diagram) from a gaml model. A tutorial is available here.

https://gama-platform.org/wiki/G__GraphicalEditor

**Comment by mbalagueraj** [comment]
Hi, the plugin download site update site: http://updates.gama-platform.org/graphical_modeling/1.9.0 and the sites included in the Windows 11 app are down. Can I have any alternative? 
Regards

**Reply by RoiArthurB** [reply]
Hi, the server is back up and the plugin should be available again, sorry for the inconvenience. :)

On a side note, the code of this plugin can be found here: https://github.com/gama-platform/gama.graphical.modeling and you can either compile it yourself or, if you're using the git version within Eclipse, you can add it in your workspace to integrate that plugin from the sources :)

**Reply by mbalagueraj** [reply]
Thanks for your answer. http://updates.gama-platform.org/graphical_modeling/1.9.0 still down and I would like if there exist any guide in order to learn how to compile the plugin. Regards.

**Reply by RoiArthurB** [reply]
Hi again, I'm really confused as the server and the URL are working perfectly on my side... Just as a side note, this URL can't be visited with your browser but if you're using it within GAMA will display the plugin that you can install (everything's explained here: https://gama-platform.org/wiki/InstallingPlugins ) :)

If you want to compile it, either you can add it in your eclipse workspace (you should be able to find all the necessary documentation here : https://gama-platform.org/wiki/InstallingGitVersion ), or if you just want to compile this plugin, you'll need to have Java and Maven install on your computer. From there, you can simply run the command `mvn clear install` from the `idees.gama.parent/` folder inside the plugin's repository. Even if possible, I still would recommend installing the compiled version through the normal install process explained above :)

**Reply by mbalagueraj** [reply]
Here is a screenshot:
https://drive.google.com/file/d/1rrXVmVxqjJKiBwz1iXsCSaDWebWshCfC/view?usp=sharing
Thanks

**Reply by RoiArthurB** [reply]
@mbalagueraj You can find the correct URL with the dropdown menu where you typed it, your is outdated.

The correct URL is https://updates.gama-platform.org/graphical_modeling/1.9.0 (in HTTPS, not HTTP) :)

**Reply by mbalagueraj** [reply]
Now I got it, thanks very much.

**Comment by mbalagueraj** [comment]
Hi, becouse a hardware problem I had to install GAMA again, but when I try to install the graphical modeler plugin, I receive the following message:

" Cannot complete the install because one or more required items could not be found.
  Software being installed: Graphical_modeling 1.9.0.202303060245 (idees.gama.feature.graphical_modeling.feature.group 1.9.0.202303060245)
  Missing requirement: Xbase Runtime Library 2.29.0.v20221121-0915 (org.eclipse.xtext.xbase.lib 2.29.0.v20221121-0915) requires 'osgi.bundle; com.google.guava [30.1.0,31.0.0)' but it could not be found
  Cannot satisfy dependency:
    From: Graphical_modeling 1.9.0.202303060245 (idees.gama.feature.graphical_modeling.feature.group 1.9.0.202303060245)
    To: org.eclipse.equinox.p2.iu; org.eclipse.xtext.xbase.lib [2.29.0.v20221121-0915,2.29.0.v20221121-0915] "

¿Could you help me please?, Thanks

**Reply by RoiArthurB** [reply]
Hello @mbalagueraj ,

I guess this is beyond what this webpage is talking about. Please open an issue on the plugin's repository explaining your problem to receive proper support : https://github.com/gama-platform/gama.graphical.modeling :)

**Reply by mbalagueraj** [reply]
Thanks for your soon answer. I installed GAMA 1.9.2 and the plugin installed correctly.
A greeting.

**Comment by mk45** [comment]
In this tutorial is mentioned "It allows as well to produce a graphical model (diagram) from a gaml model." But I can not find any info about this feature here or elsewhere.

Is the functionality true ? Can You please direct me where such informations can be found ?

**Reply by jagodasabaj-lab** [reply]
I am also interested

**Reply by lesquoyb** [reply]
@ptaillandier any idea ?
